management_api_app/tests/test_api/test_routes/test_workspace_templates.py

```diff
@@ -38,7 +46,25 @@ async def test_post_workspace_templates_does_not_create_a_template_with_bad_payl
     response = await client.post(app.url_path_for(strings.API_CREATE_WORKSPACE_TEMPLATES), json=input_data)
 
     assert response.status_code == status.HTTP_422_UNPROCESSABLE_ENTITY
-# [GET] /workspace-templates/{template_name}
+
+
+@patch("api.routes.workspace_templates.WorkspaceTemplateRepository.get_current_workspace_template_by_name")
+async def test_when_updating_current_and_template_not_found_return_404(mock, app: FastAPI, client: AsyncClient):
+    mock.side_effect = EntityDoesNotExist
+
+    response = await client.post(app.url_path_for(strings.API_CREATE_WORKSPACE_TEMPLATES), json=payload)
+
+    assert response.status_code == status.HTTP_404_NOT_FOUND
```

Comment:
As per above... shouldn't this be ok? how do you create the first current otherwise?